src/index.ts

```diff
@@ -16,17 +24,72 @@ async function run(): Promise<void> {
     const image = ghCore.getInput(Inputs.IMAGE);
     const namespace = ghCore.getInput(Inputs.NAMESPACE);
     const port = ghCore.getInput(Inputs.PORT);
+    const createPullSecretFrom = ghCore.getInput(Inputs.CREATE_PULL_SECRET_FROM);
+    const registry = ghCore.getInput(Inputs.REGISTRY_HOSTNAME);
+    const registryUsername = ghCore.getInput(Inputs.REGISTRY_USERNAME);
+    const registryPassword = ghCore.getInput(Inputs.REGISTRY_PASSWORD);
 
     const appSelector = utils.getSelector(appName);
 
-    let namespaceArg: string | undefined;
     if (namespace) {
         namespaceArg = `--namespace=${namespace}`;
     }
     else {
         ghCore.info(`No namespace provided`);
     }
 
+    if (createPullSecretFrom === "docker") {
```

Comment:
all this pull secret logic should probably be in another function instead of top-level `run`

Reply:
this if block could be its own function returning a boolean which could be assigned to `isPullSecretCreated`